lib/Reader/readerir.cpp

```diff
@@ -2069,9 +2069,10 @@ IRNode *GenIR::convertFromStackType(IRNode *Node, CorInfoType CorType,
       Result = (IRNode *)LLVMBuilder->CreateIntCast(Node, ResultTy, IsSigned);
     } else if (NeedsExtension) {
       assert(!NeedsReinterpret && "cannot reinterpret and extend");
-      assert((CorType == CorInfoType::CORINFO_TYPE_NATIVEINT) &&
-             "only expect to extend to native int");
-      const bool IsSigned = true;
+      assert((CorType == CorInfoType::CORINFO_TYPE_NATIVEINT ||
+              CorType == CorInfoType::CORINFO_TYPE_NATIVEUINT) &&
+             "only expect to extend to native int or uint");
+      const bool IsSigned = CorType == CorInfoType::CORINFO_TYPE_NATIVEINT;
```

Comment:
RyuJIT sign-extends in both cases, but this appears to be a bug: ECMA-335 Table III.9 prescribes sign-extension in the case of native int and zero-extension in the case of native unsigned int.